python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -236,26 +246,38 @@ def _build_status_cache(
     asset_key: AssetKey,
     partitions_def: Optional[PartitionsDefinition],
     dynamic_partitions_store: DynamicPartitionsStore,
-    stored_cache_value: Optional[AssetStatusCacheValue] = None,
-    last_materialization_storage_id: Optional[int] = None,
+    stored_cache_value: Optional[AssetStatusCacheValue],
+    asset_record: Optional["AssetRecord"],
 ) -> Optional[AssetStatusCacheValue]:
     """This method refreshes the asset status cache for a given asset key. It recalculates
     the materialized partition subset for the asset key and updates the cache value.
     """
+    last_materialization_storage_id = (
+        asset_record.asset_entry.last_materialization_storage_id if asset_record else None
+    )
+
+    last_planned_materialization_storage_id = get_last_planned_storage_id(
+        instance, asset_key, asset_record
+    )
+
     latest_storage_id = max(
         last_materialization_storage_id if last_materialization_storage_id else 0,
-        _get_last_planned_storage_id(instance, asset_key),
+        last_planned_materialization_storage_id if last_planned_materialization_storage_id else 0,
```

Comment:
nit: `last_planned_materialization_storage_id or 0` does the same thing and is a bit easier to read (to my eye)